firefox-ios/Client/Frontend/TrackingProtection/BlockedTrackersTableController.swift

```diff
@@ -0,0 +1,385 @@
+
+    // MARK: Navigation View
+    // TODO: Replace this with the HeaderView defined for the TrackingDetailsViewController
+    private let navigationView = UIView()
+    private let horizontalLine: UIView = .build { _ in }
+    private let siteTitleLabel: UILabel = .build { label in
+        label.adjustsFontForContentSizeCategory = true
+        label.textAlignment = .center
+        label.font = FXFontStyles.Regular.headline.scaledFont()
+        label.numberOfLines = 2
+        label.accessibilityTraits.insert(.header)
+    }
+    private var closeButton: UIButton = .build { button in
+        button.layer.cornerRadius = 0.5 * TPMenuUX.UX.closeButtonSize
+        button.clipsToBounds = true
+        button.imageView?.contentMode = .scaleAspectFit
+        button.setImage(UIImage(named: StandardImageIdentifiers.Medium.cross), for: .normal)
+    }
+    private var backButton: UIButton = .build { button in
+        button.layer.cornerRadius = 0.5 * TPMenuUX.UX.closeButtonSize
+        button.clipsToBounds = true
+        button.setTitle("Back", for: .normal)
+        button.setImage(UIImage(imageLiteralResourceName: StandardImageIdentifiers.Large.chevronLeft)
+            .withRenderingMode(.alwaysTemplate),
+                        for: .normal)
+        button.titleLabel?.font = TPMenuUX.Fonts.viewTitleLabels.scaledFont()
+    }
+
+    private var constraints = [NSLayoutConstraint]()
+    var model: BlockedTrackersTableModel
+    var notificationCenter: NotificationProtocol
+    var themeManager: ThemeManager
+    var themeObserver: NSObjectProtocol?
+    let windowUUID: WindowUUID
+
+    var currentWindowUUID: UUID? { return windowUUID }
+
+    init(with model: BlockedTrackersTableModel,
+         windowUUID: WindowUUID,
+         and notificationCenter: NotificationProtocol = NotificationCenter.default,
+         themeManager: ThemeManager = AppContainer.shared.resolve()) {
+        self.model = model
+        self.windowUUID = windowUUID
+        self.notificationCenter = notificationCenter
+        self.themeManager = themeManager
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        notificationCenter.removeObserver(self)
+    }
+
+    // MARK: View Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        setupDataSource()
+        applySnapshot()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        updateViewDetails()
+        listenForThemeChange(view)
+        applyTheme()
+    }
+
+    // MARK: View Setup
+    private func setupView() {
+        setupHeaderView()
+        setupTableView()
+        NSLayoutConstraint.activate(constraints)
+    }
+
+    // MARK: Header View Setup
+    private func setupHeaderView() {
```

Comment:
if we have a separate header view class, why not add most of these constraints in that file instead, similar to `MicrosurveyTableHeaderView`, this can remove the logic for the VC (doesn't need to know about it) and we can clearly see that everything is related to the header view?

Reply:
this is already done in the [Details Screen PR](https://github.com/mozilla-mobile/firefox-ios/pull/21527), waiting for it to be approved so I can use that view instead.